src/WebJobs.Script/SanitizerRegex.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Azure.WebJobs.Logging
+{
+    internal partial class SanitizerRegex
+    {
+        // Pattern of format : "<protocol>://<username>:<password>@<address>:<port>"
+        private static readonly string Pattern = @"
+                                                \b([a-zA-Z]+)            # Capture protocol
+                                                :\/\/                    # '://'
+                                                ([^:/\s]+)               # Capture username
+                                                :                        # ':'
+                                                ([^@/\s]+)               # Capture password
+                                                @                        # '@'
+                                                ([^:/\s]+)               # Capture address
+                                                :                        # ':'
+                                                ([0-9]+)\b               # Capture port number
+                                            ";
+
+#if NET8_0_OR_GREATER
+        [GeneratedRegex(Pattern, RegexOptions.IgnoreCase | RegexOptions.IgnorePatternWhitespace, "en-US")]
```

Comment:
Is this code buildable ? Don't we need to add `<TargetFramework>net8.0</TargetFramework>` to the csproj for this to work?

Reply:
The code could be buildable, but would it throw during runtime due to the other error I mentioned about using `readonly`

Reply:
Yes this wont build as of now. When we add TFM to net8 and move all other regex in our repo to use this feature then this one will automatically work. Or may be I can remove using this feature for now. I want to keep scope of this PR only to sanitize the log.